daphne_worker/src/dap.rs

```diff
@@ -64,29 +65,90 @@ pub(crate) fn dap_response_to_worker(resp: DapResponse) -> Result<Response> {
     Ok(worker_resp)
 }
 
+#[async_trait(?Send)]
 impl<D> HpkeDecrypter for DaphneWorkerConfig<D> {
-    fn get_hpke_config_for(&self, _task_id: &Id) -> Option<&HpkeConfig> {
-        if self.hpke_receiver_config_list.is_empty() {
-            return None;
-        }
+    async fn get_hpke_config_for(
+        &self,
+        _task_id: &Id,
+    ) -> std::result::Result<Option<HpkeConfig>, DapError> {
+        let kv_store = self.kv("DAP_HPKE_RECEIVER_CONFIG_STORE").map_err(dap_err)?;
+        let keys = kv_store
+            .list()
+            .execute()
+            .await
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
 
-        // Always advertise the first HPKE config in the list.
-        Some(&(self.hpke_receiver_config_list[0]).config)
+        if keys.keys.is_empty() {
+            // Generate a new random HPKE config ID.
+            let hpke_config_id: u8 = rand::thread_rng().gen();
+            let name = format!("hpke_receiver_config/{}", hpke_config_id);
```

Comment:
Because "hpke_reciever_config" is used in multiple places, it's a good idea to declare it as a constant. At the top of this file:
```rust
const KV_KEY_PREFIX_HPKE_RECEIVER_CONFIG: &'str = "hpke_receiver_config";
```

Then here we would have
```rust
let name = format!("{}/{}", KV_KEY_PREFIX_HPKE_RECEIVER_CONFIG, hpke_config_id);
```